File: src/main/java/com/example/swcproject/Message.java
package com.example.swcproject;

public class Message {

    private final String message;
    private final MESSAGE_TYPE type;

    public enum MESSAGE_TYPE {
        INFO,
        WARNING,
        ERROR
    }

    public Message(String message, MESSAGE_TYPE type) {
        this.message = message;
        this.type = type;
    }

    public String getMessage() {
        return message;
    }

    public MESSAGE_TYPE getType() {
        return type;
    }

    @Override
    public String toString() {
        return  type + ": " + message;
    }
}

File: src/main/scala/com/example/scala/Message.scala
package com.example.scala

import com.example.scala.MessageType.MessageType

class Message(private val message: String, val messageType: MessageType) {
  //def getMessage(): String = message
  //def getType(): MessageType = messageType

  override def toString: String = s"$messageType: $message"
}

object MessageType extends Enumeration {
  type MessageType = Value
  val INFO, WARNING, ERROR = Value
}